.github/workflows/lint.yml

```diff
@@ -0,0 +1,23 @@
+name: Lint
+
+on: [push]
+
+jobs:
+  flake8-lint:
+    runs-on: ubuntu-latest
+    name: flake8 Lint
+
+    steps:
+      - name: Check out source repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python environment
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Install flake8
+        run: pip install flake8
+
+      - name: Run flake8
+        run: flake8 .
```

Comment:
I would not add flake8 since [ruff](https://github.com/astral-sh/ruff) is a thing these days.

Besides, adding linting sounds like a different PR than just getting GHA running?

Reply:
Oh, not to mention the repo already uses `pre-commit`, so you can just use `pre-commit/action` :)

https://github.com/YatingMusic/miditoolkit/blob/6f0a8f3a78885e0b6614c70c097230f47a292361/.github/workflows/lint.yml#L1-L17

Reply:
I'm a bit confused. Are you suggesting to keep using pre-commit instead of ruff?

Reply:
Okay. I see

Reply:
There's a bit more to it. Jazzband enforce pre-commit as a CI step eg:// https://results.pre-commit.ci/run/github/1360247/1701345236.N6ZJwVBqR5qkswtYyt4tsw

So we must have some sort of pre-commit tooling. Running the precommit tooling against file changes in the diff is the most consistent way of enforcing linting.

Switching to Ruff looks fine. It looks easy to back out of in the unlikely event there's issues with it